File: python_programmers_school/level 1/행렬의 덧셈.py
def solution(arr1, arr2):
    res = []
    for i in range(len(arr1)):
        n = []
        for j in range(len(arr1[i])):
            n.append(arr1[i][j] + arr2[i][j])
        res.append(n)
    return res

# 다른 풀이
# def sumMatrix(A,B):
#     answer = [[c + d for c, d in zip(a,b)] for a, b in zip(A,B)]
#     return answer
# zip(): 여러 개의 순회 가능한(iterable) 객체를 인자로 받고, 각 객체가 담고 있는 원소를 튜플의 형태로 차례로 접근할 수 
# 있는 반복자(iterator)를 반환 (A와 B에는 arr1, arr2가 들어가고, a와 b에는 (arr1[i], arr2[i]) 이렇게 들어감
# 그리고 c, d에는 (arr1[i][j], arr2[i][j])가 들어가서 그거 2개가 더해진 값을 이차원 리스트 안에 넣음.)

File: javascript_programmers_school/level 1/행렬의 덧셈.js
function solution(arr1, arr2) {
  let answer = [];
  for (i in arr1) {
    let tmp = [];
    for (j in arr1[i]) {
      tmp.push(arr1[i][j] + arr2[i][j]);
    }
    answer.push(tmp);
  }
  return answer;
}

// 다른 사람의 풀이 (map 이용, A는 arr1, B는 arr2라고 생각하면 됨.)
function solution(A, B) {
  return A.map((arr1, idx1) => arr1.map((val, idx2) => val + B[idx1][idx2]));
}
// arr1의 값을 B의 똑같은 인덱스의 값을 더한 값으로 업데이트하는 식으로 진행
